rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php

```diff
@@ -228,4 +235,31 @@ private function shouldSkipArrayCallable(Class_ $class, ?ArrayCallable $arrayCal
         // is current class method?
         return ! $this->isName($class, $arrayCallable->getClass());
     }
+
+    private function isPrivateAbstractMethodInTrait(ClassMethod $classMethod, ?string $classMethodName): bool
+    {
+        if (null === $classMethodName) {
+            return false;
+        }
+
+        $scope = $classMethod->getAttribute(AttributeKey::SCOPE);
+        if (! $scope instanceof Scope) {
+            return false;
+        }
```

Comment:
scope check no longer needed as already checked in shouldSkip()

Reply:
Not sure about this to be honest.

I agree, it is checked in  `shouldSkip` but this would imply a direct dependency between `shouldSkip` and `isPrivateAbstractMethodInTrait` (which is not directly visible imo).

Is this done in the rest of rector-src, too? If so I'm happy to change it :)

Reply:
Or would it be worth it to create a new method like `getClassReflection` that returns the ClassReflection or null (on failure), that way we can keep the "checks in sync" if that makes sense.

Reply:
I think you can pass ClassReflection so no need double check, I think changing private method parameter is fine

Reply:
Since Class complexity got over 30 I had to move the "is used" checks into it's own class.

This results in the `shouldSkip` function being completely seperated from the "is used" checks which makes the if condition([s](https://github.com/rectorphp/rector-src/pull/604#discussion_r683842138)) useful again.

(This is why I wanted to keep them in the first place to ensure that in future refactorings/seperations this wouldn't break :p)